<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Theory &amp; Methods ‚Ä¢ scMetaLink</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Theory &amp; Methods">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">scMetaLink</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/01-quick-start.html">Quick Start Guide</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/02-theory.html">Theory &amp; Methods</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/03-production-sensing.html">Production &amp; Sensing Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/04-communication.html">Communication Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/05-visualization.html">Visualization Gallery</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/06-applications.html">Application Examples</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/Zaoqu-Liu/scMetaLink" aria-label="GitHub"><span class="fa fab fa-github"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Theory &amp; Methods</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/Zaoqu-Liu/scMetaLink/blob/HEAD/vignettes/02-theory.Rmd" class="external-link"><code>vignettes/02-theory.Rmd</code></a></small>
      <div class="d-none name"><code>02-theory.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>scMetaLink models metabolite-mediated cell communication as a
two-step process:</p>
<div class="figure" style="text-align: center">
<img src="flowchart.png" alt="**Figure 1: scMetaLink Workflow.** Schematic overview of the analysis pipeline for inferring metabolite-mediated cell-cell communication." width="100%"><p class="caption">
<strong>Figure 1: scMetaLink Workflow.</strong> Schematic overview of
the analysis pipeline for inferring metabolite-mediated cell-cell
communication.
</p>
</div>
</div>
<div class="section level2">
<h2 id="the-metalinksdb-knowledge-base">The MetalinksDB Knowledge Base<a class="anchor" aria-label="anchor" href="#the-metalinksdb-knowledge-base"></a>
</h2>
<p>scMetaLink uses <strong>MetalinksDB</strong>, a comprehensive
database of metabolite-protein interactions.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Zaoqu-Liu/scMetaLink" class="external-link">scMetaLink</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load and explore the database</span></span>
<span><span class="va">db</span> <span class="op">&lt;-</span> <span class="fu">scMetaLink</span><span class="fu">:::</span><span class="fu"><a href="../reference/dot-load_metalinksdb.html">.load_metalinksdb</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"=== MetalinksDB Statistics ===\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; === MetalinksDB Statistics ===</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Metabolites:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">db</span><span class="op">$</span><span class="va">metabolites</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Metabolites: 1128</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Proteins:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">db</span><span class="op">$</span><span class="va">proteins</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Proteins: 4374</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Interactions:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">db</span><span class="op">$</span><span class="va">edges</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Interactions: 41894</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Pathways:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">db</span><span class="op">$</span><span class="va">pathway</span><span class="op">$</span><span class="va">pathway</span><span class="op">)</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Pathways: 36806</span></span></code></pre></div>
<div class="section level3">
<h3 id="interaction-types">Interaction Types<a class="anchor" aria-label="anchor" href="#interaction-types"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Two main interaction types</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"=== Interaction Types ===\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; === Interaction Types ===</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">db</span><span class="op">$</span><span class="va">edges</span><span class="op">$</span><span class="va">type</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    lr    pd </span></span>
<span><span class="co">#&gt; 11869 30025</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"\n- lr (ligand-receptor): Direct metabolite-receptor binding\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; - lr (ligand-receptor): Direct metabolite-receptor binding</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"- pd (produce-degrade): Enzymatic production/degradation\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; - pd (produce-degrade): Enzymatic production/degradation</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="mode-of-regulation-mor">Mode of Regulation (MOR)<a class="anchor" aria-label="anchor" href="#mode-of-regulation-mor"></a>
</h3>
<p>For <code>pd</code> type interactions, MOR indicates the
direction:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pd_edges</span> <span class="op">&lt;-</span> <span class="va">db</span><span class="op">$</span><span class="va">edges</span><span class="op">[</span><span class="va">db</span><span class="op">$</span><span class="va">edges</span><span class="op">$</span><span class="va">type</span> <span class="op">==</span> <span class="st">"pd"</span>, <span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"=== Mode of Regulation for pd interactions ===\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; === Mode of Regulation for pd interactions ===</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">pd_edges</span><span class="op">$</span><span class="va">mor</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    -1     1 </span></span>
<span><span class="co">#&gt; 13144 16881</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"\n- mor = 1: Enzyme produces/secretes the metabolite\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; - mor = 1: Enzyme produces/secretes the metabolite</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"- mor = -1: Enzyme degrades/consumes the metabolite\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; - mor = -1: Enzyme degrades/consumes the metabolite</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"- mor = 0: Enzyme binds without direction change\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; - mor = 0: Enzyme binds without direction change</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="protein-types">Protein Types<a class="anchor" aria-label="anchor" href="#protein-types"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"=== Protein Classifications ===\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; === Protein Classifications ===</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">db</span><span class="op">$</span><span class="va">proteins</span><span class="op">$</span><span class="va">protein_type</span>, useNA <span class="op">=</span> <span class="st">"ifany"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; catalytic_receptor             enzyme               gpcr               lgic </span></span>
<span><span class="co">#&gt;                179                918                306                 71 </span></span>
<span><span class="co">#&gt;                nhr           other_ic      other_protein        transporter </span></span>
<span><span class="co">#&gt;                 43                 39                 28                393 </span></span>
<span><span class="co">#&gt;               vgic               &lt;NA&gt; </span></span>
<span><span class="co">#&gt;                 73               2324</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Receptors: gpcr, lgic, nhr, vgic, catalytic_receptor, other_ic\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Receptors: gpcr, lgic, nhr, vgic, catalytic_receptor, other_ic</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Enzymes: enzyme (or NA in protein_type)\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Enzymes: enzyme (or NA in protein_type)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Transporters: transporter\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Transporters: transporter</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="mathematical-framework">Mathematical Framework<a class="anchor" aria-label="anchor" href="#mathematical-framework"></a>
</h2>
<div class="section level3">
<h3 id="metabolite-production-potential-mpp">1. Metabolite Production Potential (MPP)<a class="anchor" aria-label="anchor" href="#metabolite-production-potential-mpp"></a>
</h3>
<p>The production potential for metabolite
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
in cell type
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>c</mi><annotation encoding="application/x-tex">c</annotation></semantics></math>
is:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mtext mathvariant="normal">MPP</mtext><mrow><mi>m</mi><mo>,</mo><mi>c</mi></mrow></msub><mo>=</mo><mfrac><mn>1</mn><mrow><mo stretchy="false" form="prefix">|</mo><msub><mi>E</mi><mi>m</mi></msub><mo stretchy="false" form="prefix">|</mo></mrow></mfrac><munder><mo>‚àë</mo><mrow><mi>e</mi><mo>‚àà</mo><msub><mi>E</mi><mi>m</mi></msub></mrow></munder><mtext mathvariant="normal">Score</mtext><mo stretchy="false" form="prefix">(</mo><mi>e</mi><mo>,</mo><mi>c</mi><mo stretchy="false" form="postfix">)</mo><mo>‚àí</mo><mi>Œ±</mi><mo>‚ãÖ</mo><mfrac><mn>1</mn><mrow><mo stretchy="false" form="prefix">|</mo><msub><mi>D</mi><mi>m</mi></msub><mo stretchy="false" form="prefix">|</mo></mrow></mfrac><munder><mo>‚àë</mo><mrow><mi>d</mi><mo>‚àà</mo><msub><mi>D</mi><mi>m</mi></msub></mrow></munder><mtext mathvariant="normal">Score</mtext><mo stretchy="false" form="prefix">(</mo><mi>d</mi><mo>,</mo><mi>c</mi><mo stretchy="false" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">\text{MPP}_{m,c} = \frac{1}{|E_m|} \sum_{e \in E_m} \text{Score}(e, c) - \alpha \cdot \frac{1}{|D_m|} \sum_{d \in D_m} \text{Score}(d, c)</annotation></semantics></math></p>
<p>Where: -
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>E</mi><mi>m</mi></msub><annotation encoding="application/x-tex">E_m</annotation></semantics></math>:
Set of enzymes that produce metabolite
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
-
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>D</mi><mi>m</mi></msub><annotation encoding="application/x-tex">D_m</annotation></semantics></math>:
Set of enzymes that degrade metabolite
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
-
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>Œ±</mi><mo>=</mo><mn>0.5</mn></mrow><annotation encoding="application/x-tex">\alpha = 0.5</annotation></semantics></math>:
Degradation weight factor -
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mtext mathvariant="normal">Score</mtext><mo stretchy="false" form="prefix">(</mo><mi>g</mi><mo>,</mo><mi>c</mi><mo stretchy="false" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">\text{Score}(g, c)</annotation></semantics></math>:
Gene expression score for gene
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>g</mi><annotation encoding="application/x-tex">g</annotation></semantics></math>
in cell type
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>c</mi><annotation encoding="application/x-tex">c</annotation></semantics></math></p>
<div class="section level4">
<h4 id="gene-expression-scoring">Gene Expression Scoring<a class="anchor" aria-label="anchor" href="#gene-expression-scoring"></a>
</h4>
<p>Three methods are available:</p>
<table class="table">
<thead><tr class="header">
<th>Method</th>
<th>Formula</th>
<th>Use Case</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>mean</code></td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mover><mi>x</mi><mo accent="true">‚Äæ</mo></mover><mi>g</mi></msub><annotation encoding="application/x-tex">\bar{x}_g</annotation></semantics></math></td>
<td>Simple average expression</td>
</tr>
<tr class="even">
<td><code>proportion</code></td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>P</mi><mo stretchy="false" form="prefix">(</mo><msub><mi>x</mi><mi>g</mi></msub><mo>&gt;</mo><mn>0</mn><mo stretchy="false" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">P(x_g &gt; 0)</annotation></semantics></math></td>
<td>Proportion of expressing cells</td>
</tr>
<tr class="odd">
<td><code>combined</code></td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mover><mi>x</mi><mo accent="true">‚Äæ</mo></mover><mi>g</mi></msub><mo>√ó</mo><mi>P</mi><mo stretchy="false" form="prefix">(</mo><msub><mi>x</mi><mi>g</mi></msub><mo>&gt;</mo><mn>0</mn><mo stretchy="false" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">\bar{x}_g \times P(x_g &gt; 0)</annotation></semantics></math></td>
<td>Balanced (recommended)</td>
</tr>
</tbody>
</table>
<table class="table">
<thead><tr class="header">
<th align="left">Method</th>
<th align="left">Formula</th>
<th align="right">Value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">mean</td>
<td align="left">mean(expr)</td>
<td align="right">0.610</td>
</tr>
<tr class="even">
<td align="left">proportion</td>
<td align="left">mean(expr &gt; 0)</td>
<td align="right">0.300</td>
</tr>
<tr class="odd">
<td align="left">combined</td>
<td align="left">mean x proportion</td>
<td align="right">0.183</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="trimean-option">Trimean Option<a class="anchor" aria-label="anchor" href="#trimean-option"></a>
</h4>
<p>For single-cell data with high dropout, <strong>trimean</strong>
provides a robust alternative:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mtext mathvariant="normal">Trimean</mtext><mo>=</mo><mfrac><mrow><msub><mi>Q</mi><mn>1</mn></msub><mo>+</mo><mn>2</mn><msub><mi>Q</mi><mn>2</mn></msub><mo>+</mo><msub><mi>Q</mi><mn>3</mn></msub></mrow><mn>4</mn></mfrac></mrow><annotation encoding="application/x-tex">\text{Trimean} = \frac{Q_1 + 2Q_2 + Q_3}{4}</annotation></semantics></math></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example: Trimean vs Arithmetic Mean</span></span>
<span><span class="va">expr_with_outlier</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">100</span><span class="op">)</span> <span class="co"># Outlier = 100</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Data:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">expr_with_outlier</span>, collapse <span class="op">=</span> <span class="st">", "</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Data: 0, 0, 0, 1, 2, 3, 100</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Arithmetic mean:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">expr_with_outlier</span><span class="op">)</span>, <span class="fl">2</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Arithmetic mean: 15.14</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Median:"</span>, <span class="fu"><a href="https://rdrr.io/r/stats/median.html" class="external-link">median</a></span><span class="op">(</span><span class="va">expr_with_outlier</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Median: 1</span></span>
<span></span>
<span><span class="co"># Trimean calculation</span></span>
<span><span class="va">q</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">expr_with_outlier</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.25</span>, <span class="fl">0.5</span>, <span class="fl">0.75</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">trimean</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">q</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="op">+</span> <span class="fl">2</span> <span class="op">*</span> <span class="va">q</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> <span class="op">+</span> <span class="va">q</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span> <span class="op">/</span> <span class="fl">4</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Trimean:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">trimean</span>, <span class="fl">2</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Trimean: 1.12</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"\nTrimean is robust to the outlier!\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Trimean is robust to the outlier!</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="metabolite-sensing-capability-msc">2. Metabolite Sensing Capability (MSC)<a class="anchor" aria-label="anchor" href="#metabolite-sensing-capability-msc"></a>
</h3>
<p>The sensing capability for metabolite
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
in cell type
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>c</mi><annotation encoding="application/x-tex">c</annotation></semantics></math>
is:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mtext mathvariant="normal">MSC</mtext><mrow><mi>m</mi><mo>,</mo><mi>c</mi></mrow></msub><mo>=</mo><mfrac><mn>1</mn><mrow><mo stretchy="false" form="prefix">|</mo><msub><mi>R</mi><mi>m</mi></msub><mo stretchy="false" form="prefix">|</mo></mrow></mfrac><munder><mo>‚àë</mo><mrow><mi>r</mi><mo>‚àà</mo><msub><mi>R</mi><mi>m</mi></msub></mrow></munder><msub><mi>w</mi><mi>r</mi></msub><mo>‚ãÖ</mo><mtext mathvariant="normal">Score</mtext><mo stretchy="false" form="prefix">(</mo><mi>r</mi><mo>,</mo><mi>c</mi><mo stretchy="false" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">\text{MSC}_{m,c} = \frac{1}{|R_m|} \sum_{r \in R_m} w_r \cdot \text{Score}(r, c)</annotation></semantics></math></p>
<p>Where: -
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>R</mi><mi>m</mi></msub><annotation encoding="application/x-tex">R_m</annotation></semantics></math>:
Set of receptors/transporters for metabolite
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
-
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>w</mi><mi>r</mi></msub><annotation encoding="application/x-tex">w_r</annotation></semantics></math>:
Affinity weight based on interaction confidence score</p>
<div class="section level4">
<h4 id="affinity-weighting">Affinity Weighting<a class="anchor" aria-label="anchor" href="#affinity-weighting"></a>
</h4>
<p>Receptor-metabolite binding affinities from MetalinksDB are used as
weights:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lr_edges</span> <span class="op">&lt;-</span> <span class="va">db</span><span class="op">$</span><span class="va">edges</span><span class="op">[</span><span class="va">db</span><span class="op">$</span><span class="va">edges</span><span class="op">$</span><span class="va">type</span> <span class="op">==</span> <span class="st">"lr"</span>, <span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Affinity score distribution (0-1000):\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Affinity score distribution (0-1000):</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">lr_edges</span><span class="op">$</span><span class="va">combined_score</span><span class="op">)</span></span>
<span><span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's </span></span>
<span><span class="co">#&gt;   202.0   287.0   413.0   553.5   921.0   999.0     470</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="hill-function-optional">Hill Function (Optional)<a class="anchor" aria-label="anchor" href="#hill-function-optional"></a>
</h4>
<p>To model receptor saturation kinetics, an optional Hill
transformation can be applied:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>P</mi><mo>=</mo><mfrac><msup><mi>E</mi><mi>n</mi></msup><mrow><msubsup><mi>K</mi><mi>h</mi><mi>n</mi></msubsup><mo>+</mo><msup><mi>E</mi><mi>n</mi></msup></mrow></mfrac></mrow><annotation encoding="application/x-tex">P = \frac{E^n}{K_h^n + E^n}</annotation></semantics></math></p>
<p>Where: -
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>E</mi><annotation encoding="application/x-tex">E</annotation></semantics></math>:
Expression level (normalized) -
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>n</mi><annotation encoding="application/x-tex">n</annotation></semantics></math>:
Hill coefficient (cooperativity) -
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>K</mi><mi>h</mi></msub><annotation encoding="application/x-tex">K_h</annotation></semantics></math>:
Half-maximal threshold</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Hill function visualization</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">0.01</span><span class="op">)</span></span>
<span><span class="va">hill</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">n</span>, <span class="va">Kh</span><span class="op">)</span> <span class="va">x</span><span class="op">^</span><span class="va">n</span> <span class="op">/</span> <span class="op">(</span><span class="va">Kh</span><span class="op">^</span><span class="va">n</span> <span class="op">+</span> <span class="va">x</span><span class="op">^</span><span class="va">n</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">x</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"l"</span>, lty <span class="op">=</span> <span class="fl">2</span>, col <span class="op">=</span> <span class="st">"gray"</span>,</span>
<span>  xlab <span class="op">=</span> <span class="st">"Expression (normalized)"</span>, ylab <span class="op">=</span> <span class="st">"Response"</span>,</span>
<span>  main <span class="op">=</span> <span class="st">"Hill Function Transformation"</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/lines.html" class="external-link">lines</a></span><span class="op">(</span><span class="va">x</span>, <span class="fu">hill</span><span class="op">(</span><span class="va">x</span>, n <span class="op">=</span> <span class="fl">1</span>, Kh <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"blue"</span>, lwd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/lines.html" class="external-link">lines</a></span><span class="op">(</span><span class="va">x</span>, <span class="fu">hill</span><span class="op">(</span><span class="va">x</span>, n <span class="op">=</span> <span class="fl">2</span>, Kh <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"red"</span>, lwd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/lines.html" class="external-link">lines</a></span><span class="op">(</span><span class="va">x</span>, <span class="fu">hill</span><span class="op">(</span><span class="va">x</span>, n <span class="op">=</span> <span class="fl">0.5</span>, Kh <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"green"</span>, lwd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/legend.html" class="external-link">legend</a></span><span class="op">(</span><span class="st">"bottomright"</span>,</span>
<span>  legend <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Linear"</span>, <span class="st">"n=1 (Michaelis-Menten)"</span>, <span class="st">"n=2 (Cooperative)"</span>, <span class="st">"n=0.5"</span><span class="op">)</span>,</span>
<span>  col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gray"</span>, <span class="st">"blue"</span>, <span class="st">"red"</span>, <span class="st">"green"</span><span class="op">)</span>,</span>
<span>  lty <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span>, lwd <span class="op">=</span> <span class="fl">2</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span>h <span class="op">=</span> <span class="fl">0.5</span>, v <span class="op">=</span> <span class="fl">0.5</span>, lty <span class="op">=</span> <span class="fl">3</span>, col <span class="op">=</span> <span class="st">"gray"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="02-theory_files/figure-html/hill_demo-1.png" alt="**Figure 1: Hill Function Transformation.** Different Hill coefficients (n) produce different response curves. n=1 gives standard Michaelis-Menten kinetics. n&gt;1 indicates positive cooperativity (steeper curve). The half-maximal response occurs at x=Kh." width="100%"><p class="caption">
<strong>Figure 1: Hill Function Transformation.</strong> Different Hill
coefficients (n) produce different response curves. n=1 gives standard
Michaelis-Menten kinetics. n&gt;1 indicates positive cooperativity
(steeper curve). The half-maximal response occurs at x=Kh.
</p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="communication-score">3. Communication Score<a class="anchor" aria-label="anchor" href="#communication-score"></a>
</h3>
<p>The communication score from sender
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>s</mi><annotation encoding="application/x-tex">s</annotation></semantics></math>
to receiver
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>r</mi><annotation encoding="application/x-tex">r</annotation></semantics></math>
via metabolite
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msubsup><mtext mathvariant="normal">Comm</mtext><mrow><mi>s</mi><mo>‚Üí</mo><mi>r</mi></mrow><mi>m</mi></msubsup><mo>=</mo><mi>f</mi><mo stretchy="false" form="prefix">(</mo><msub><mtext mathvariant="normal">MPP</mtext><mrow><mi>m</mi><mo>,</mo><mi>s</mi></mrow></msub><mo>,</mo><msub><mtext mathvariant="normal">MSC</mtext><mrow><mi>m</mi><mo>,</mo><mi>r</mi></mrow></msub><mo stretchy="false" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">\text{Comm}_{s \to r}^m = f(\text{MPP}_{m,s}, \text{MSC}_{m,r})</annotation></semantics></math></p>
<p>Three aggregation methods are available:</p>
<table class="table">
<colgroup>
<col width="27%">
<col width="31%">
<col width="41%">
</colgroup>
<thead><tr class="header">
<th>Method</th>
<th>Formula</th>
<th>Properties</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>geometric</code></td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msqrt><mrow><mtext mathvariant="normal">MPP</mtext><mo>√ó</mo><mtext mathvariant="normal">MSC</mtext></mrow></msqrt><annotation encoding="application/x-tex">\sqrt{\text{MPP} \times \text{MSC}}</annotation></semantics></math></td>
<td>Balanced, penalizes imbalance</td>
</tr>
<tr class="even">
<td><code>product</code></td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mtext mathvariant="normal">MPP</mtext><mo>√ó</mo><mtext mathvariant="normal">MSC</mtext></mrow><annotation encoding="application/x-tex">\text{MPP} \times \text{MSC}</annotation></semantics></math></td>
<td>Emphasizes strong bilateral signals</td>
</tr>
<tr class="odd">
<td><code>harmonic</code></td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mfrac><mrow><mn>2</mn><mo>√ó</mo><mtext mathvariant="normal">MPP</mtext><mo>√ó</mo><mtext mathvariant="normal">MSC</mtext></mrow><mrow><mtext mathvariant="normal">MPP</mtext><mo>+</mo><mtext mathvariant="normal">MSC</mtext></mrow></mfrac><annotation encoding="application/x-tex">\frac{2 \times \text{MPP} \times \text{MSC}}{\text{MPP} + \text{MSC}}</annotation></semantics></math></td>
<td>Strongly penalizes imbalance</td>
</tr>
</tbody>
</table>
<div class="figure" style="text-align: center">
<img src="02-theory_files/figure-html/comm_methods-1.png" alt="**Figure 2: Communication Score Methods.** Heat maps showing how different methods combine production (x-axis) and sensing (y-axis) scores. Geometric mean (left) provides balanced weighting. Product (center) emphasizes strong bilateral signals. Harmonic mean (right) penalizes imbalanced communication." width="100%"><p class="caption">
<strong>Figure 2: Communication Score Methods.</strong> Heat maps
showing how different methods combine production (x-axis) and sensing
(y-axis) scores. Geometric mean (left) provides balanced weighting.
Product (center) emphasizes strong bilateral signals. Harmonic mean
(right) penalizes imbalanced communication.
</p>
</div>
</div>
<div class="section level3">
<h3 id="population-size-correction-optional">4. Population Size Correction (Optional)<a class="anchor" aria-label="anchor" href="#population-size-correction-optional"></a>
</h3>
<p>When <code>population.size = TRUE</code>, communication is weighted
by cell type abundance:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msubsup><mtext mathvariant="normal">Comm</mtext><mrow><mi>s</mi><mo>‚Üí</mo><mi>r</mi></mrow><mtext mathvariant="normal">weighted</mtext></msubsup><mo>=</mo><msub><mtext mathvariant="normal">Comm</mtext><mrow><mi>s</mi><mo>‚Üí</mo><mi>r</mi></mrow></msub><mo>√ó</mo><msqrt><mrow><mfrac><msub><mi>n</mi><mi>s</mi></msub><mi>N</mi></mfrac><mo>√ó</mo><mfrac><msub><mi>n</mi><mi>r</mi></msub><mi>N</mi></mfrac></mrow></msqrt></mrow><annotation encoding="application/x-tex">\text{Comm}_{s \to r}^{\text{weighted}} = \text{Comm}_{s \to r} \times \sqrt{\frac{n_s}{N} \times \frac{n_r}{N}}</annotation></semantics></math></p>
<p>Where
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>n</mi><mi>s</mi></msub><mo>,</mo><msub><mi>n</mi><mi>r</mi></msub></mrow><annotation encoding="application/x-tex">n_s, n_r</annotation></semantics></math>
are cell counts and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>N</mi><annotation encoding="application/x-tex">N</annotation></semantics></math>
is total cells.</p>
<p><strong>Rationale</strong>: Larger cell populations contribute more
to tissue-level signaling.</p>
</div>
</div>
<div class="section level2">
<h2 id="statistical-framework">Statistical Framework<a class="anchor" aria-label="anchor" href="#statistical-framework"></a>
</h2>
<div class="section level3">
<h3 id="permutation-test">Permutation Test<a class="anchor" aria-label="anchor" href="#permutation-test"></a>
</h3>
<p>To assess significance, cell type labels are randomly shuffled:</p>
<ol style="list-style-type: decimal">
<li>Shuffle cell type labels
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>B</mi><annotation encoding="application/x-tex">B</annotation></semantics></math>
times (default: 100-1000)</li>
<li>Recalculate production, sensing, and communication for each
permutation</li>
<li>Compute empirical p-value:</li>
</ol>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>p</mi><mo>=</mo><mfrac><mrow><mn>1</mn><mo>+</mo><munderover><mo>‚àë</mo><mrow><mi>b</mi><mo>=</mo><mn>1</mn></mrow><mi>B</mi></munderover><mn>ùüô</mn><mo stretchy="false" form="prefix">[</mo><msub><mtext mathvariant="normal">Comm</mtext><mi>b</mi></msub><mo>‚â•</mo><msub><mtext mathvariant="normal">Comm</mtext><mtext mathvariant="normal">obs</mtext></msub><mo stretchy="false" form="postfix">]</mo></mrow><mrow><mi>B</mi><mo>+</mo><mn>1</mn></mrow></mfrac></mrow><annotation encoding="application/x-tex">p = \frac{1 + \sum_{b=1}^{B} \mathbb{1}[\text{Comm}_b \geq \text{Comm}_{\text{obs}}]}{B + 1}</annotation></semantics></math></p>
</div>
<div class="section level3">
<h3 id="multiple-testing-correction">Multiple Testing Correction<a class="anchor" aria-label="anchor" href="#multiple-testing-correction"></a>
</h3>
<p>Three strategies for handling multiple comparisons:</p>
<table class="table">
<colgroup>
<col width="38%">
<col width="61%">
</colgroup>
<thead><tr class="header">
<th>Method</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>metabolite_stratified</code></td>
<td>BH correction within each metabolite (recommended)</td>
</tr>
<tr class="even">
<td><code>BH</code></td>
<td>Global Benjamini-Hochberg</td>
</tr>
<tr class="odd">
<td><code>bonferroni</code></td>
<td>Global Bonferroni (very conservative)</td>
</tr>
</tbody>
</table>
<p><strong>Why <code>metabolite_stratified</code>?</strong></p>
<ul>
<li>Metabolites are independent biological signals</li>
<li>Global correction is overly conservative</li>
<li>Per-metabolite correction preserves biological interpretability</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="data-processing-pipeline">Data Processing Pipeline<a class="anchor" aria-label="anchor" href="#data-processing-pipeline"></a>
</h2>
<p><img src="02-theory_files/figure-html/pipeline-1.png" alt="" width="100%" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="key-assumptions">Key Assumptions<a class="anchor" aria-label="anchor" href="#key-assumptions"></a>
</h2>
<ol style="list-style-type: decimal">
<li>
<strong>Gene expression correlates with protein activity</strong>:
Enzyme/receptor expression levels reflect functional activity</li>
<li>
<strong>Metabolites can diffuse between cells</strong>: Produced
metabolites are available for sensing by neighboring cells</li>
<li>
<strong>Cell type labels are accurate</strong>: Incorrect
annotations will affect results</li>
<li>
<strong>Adequate cell sampling</strong>: Each cell type needs
sufficient cells for robust estimation</li>
</ol>
</div>
<div class="section level2">
<h2 id="comparison-with-ligand-receptor-methods">Comparison with Ligand-Receptor Methods<a class="anchor" aria-label="anchor" href="#comparison-with-ligand-receptor-methods"></a>
</h2>
<table class="table">
<colgroup>
<col width="10%">
<col width="44%">
<col width="44%">
</colgroup>
<thead><tr class="header">
<th>Aspect</th>
<th>Ligand-Receptor (e.g., CellChat)</th>
<th>Metabolite-Mediated (scMetaLink)</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><strong>Signal type</strong></td>
<td>Proteins</td>
<td>Small molecules</td>
</tr>
<tr class="even">
<td><strong>Diffusion range</strong></td>
<td>Short (juxtacrine/paracrine)</td>
<td>Variable (paracrine/endocrine)</td>
</tr>
<tr class="odd">
<td><strong>Database</strong></td>
<td>CellChatDB, CellPhoneDB</td>
<td>MetalinksDB</td>
</tr>
<tr class="even">
<td><strong>Key genes</strong></td>
<td>Ligands, receptors</td>
<td>Enzymes, transporters, receptors</td>
</tr>
<tr class="odd">
<td><strong>Signal complexity</strong></td>
<td>Direct binding</td>
<td>Synthesis -&gt; Secretion -&gt; Sensing</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ol style="list-style-type: decimal">
<li><p>Schafer, S., et al.¬†(2023). MetalinksDB: a knowledgebase of
metabolite-centric signaling. <em>Nature Communications</em>.</p></li>
<li><p>Xiao, Z., et al.¬†(2022). Metabolite-mediated intercellular
communication in tumor microenvironment. <em>Frontiers in Cell and
Developmental Biology</em>.</p></li>
</ol>
</div>
<div class="section level2">
<h2 id="next">Next<a class="anchor" aria-label="anchor" href="#next"></a>
</h2>
<ul>
<li>
<strong><a href="03-production-sensing.html">Production &amp;
Sensing Analysis</a></strong>: Detailed parameter tuning</li>
<li>
<strong><a href="04-communication.html">Communication
Analysis</a></strong>: Advanced communication analysis</li>
</ul>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Zaoqu Liu.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
