<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Spatial Transcriptomics Analysis • scMetaLink</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Spatial Transcriptomics Analysis">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">scMetaLink</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/quick-start.html">Quick Start Guide</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/theory.html">Theory &amp; Methods</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/production-sensing.html">Production &amp; Sensing Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/communication.html">Communication Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/spatial-analysis.html">Spatial Transcriptomics</a></li>
    <li><a class="dropdown-item" href="../articles/visualization.html">Visualization Gallery</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/applications.html">Application Examples</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/Zaoqu-Liu/scMetaLink" aria-label="GitHub"><span class="fa fab fa-github"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Spatial Transcriptomics Analysis</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/Zaoqu-Liu/scMetaLink/blob/HEAD/vignettes/spatial-analysis.Rmd" class="external-link"><code>vignettes/spatial-analysis.Rmd</code></a></small>
      <div class="d-none name"><code>spatial-analysis.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>scMetaLink supports <strong>spatial transcriptomics</strong> data,
enabling the analysis of metabolite-mediated cell communication with
spatial context. This vignette demonstrates how to:</p>
<ol style="list-style-type: decimal">
<li>Create a spatial scMetaLink object</li>
<li>Compute spatially-weighted communication</li>
<li>Visualize communication patterns on tissue coordinates</li>
<li>Identify communication hotspots</li>
</ol>
</div>
<div class="section level2">
<h2 id="spatial-analysis-workflow">Spatial Analysis Workflow<a class="anchor" aria-label="anchor" href="#spatial-analysis-workflow"></a>
</h2>
<p>The following diagram illustrates the complete spatial
transcriptomics analysis workflow in scMetaLink:</p>
<div class="figure" style="text-align: center">
<img src="spatial-flow.png" alt="**Figure 1: Spatial scMetaLink Workflow.** Overview of the spatial transcriptomics analysis pipeline, from data input through spatially-weighted communication inference to visualization." width="100%"><p class="caption">
<strong>Figure 1: Spatial scMetaLink Workflow.</strong> Overview of the
spatial transcriptomics analysis pipeline, from data input through
spatially-weighted communication inference to visualization.
</p>
</div>
</div>
<div class="section level2">
<h2 id="spatial-communication-algorithm">Spatial Communication Algorithm<a class="anchor" aria-label="anchor" href="#spatial-communication-algorithm"></a>
</h2>
<p>scMetaLink incorporates spatial distance to weight cell-cell
communication. The computational framework is illustrated below:</p>
<div class="figure" style="text-align: center">
<img src="spatial-cal.png" alt="**Figure 2: Spatial Communication Algorithm.** Mathematical framework showing how spatial distances are incorporated into metabolite production, sensing, and communication score calculations." width="100%"><p class="caption">
<strong>Figure 2: Spatial Communication Algorithm.</strong> Mathematical
framework showing how spatial distances are incorporated into metabolite
production, sensing, and communication score calculations.
</p>
</div>
</div>
<div class="section level2">
<h2 id="why-spatial-analysis">Why Spatial Analysis?<a class="anchor" aria-label="anchor" href="#why-spatial-analysis"></a>
</h2>
<p>Traditional single-cell analysis treats cells as independent units,
ignoring their physical locations. However, metabolite signaling is
fundamentally spatial:</p>
<ul>
<li>
<strong>Metabolites diffuse</strong> through tissue with
distance-dependent decay</li>
<li>
<strong>Nearby cells</strong> are more likely to communicate via
metabolites</li>
<li>
<strong>Spatial organization</strong> of cell types affects
signaling patterns</li>
</ul>
<p>scMetaLink’s spatial module incorporates these biological
realities.</p>
<p><img src="spatial-analysis_files/figure-html/concept-1.png" alt="" width="100%" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="load-example-data">Load Example Data<a class="anchor" aria-label="anchor" href="#load-example-data"></a>
</h2>
<p>scMetaLink includes a colon spatial transcriptomics dataset for
demonstration.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Zaoqu-Liu/scMetaLink" class="external-link">scMetaLink</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://Matrix.R-forge.R-project.org" class="external-link">Matrix</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load spatial colon data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">st_expr</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">st_meta</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">st_scalefactors</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check the data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"=== Spatial Transcriptomics Data ===\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; === Spatial Transcriptomics Data ===</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Expression matrix:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">st_expr</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, <span class="st">"genes x"</span>, <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">st_expr</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>, <span class="st">"spots\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Expression matrix: 4284 genes x 1000 spots</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"\nSpot metadata:\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Spot metadata:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">st_meta</span><span class="op">)</span></span>
<span><span class="co">#&gt;                        x     y array_row array_col   cell_type</span></span>
<span><span class="co">#&gt; TAGTCCCGGAGACCAC-1  6642 13195        66        48 Endothelial</span></span>
<span><span class="co">#&gt; GTGCGTGTATATGAGC-1 10097  8796        41        83     Stromal</span></span>
<span><span class="co">#&gt; CCGGTATCTGGCGACT-1 10371 15069        77        85  Fibroblast</span></span>
<span><span class="co">#&gt; CCCAAGAATGCACGGT-1 10519  1993         2        88      Immune</span></span>
<span><span class="co">#&gt; CGAAACATAGATGGCA-1  9435  3574        11        77      Immune</span></span>
<span><span class="co">#&gt; TTCTTATCCGCTGGGT-1 11917 10169        49       101  Fibroblast</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"\nCell type distribution:\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Cell type distribution:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">st_meta</span><span class="op">$</span><span class="va">cell_type</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Endothelial  Epithelial  Fibroblast      Immune     Stromal       Tumor </span></span>
<span><span class="co">#&gt;         174         189         232         145         133         127</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"\nScale factors:\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Scale factors:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">st_scalefactors</span><span class="op">)</span></span>
<span><span class="co">#&gt; $spot_diameter_fullres</span></span>
<span><span class="co">#&gt; [1] 130.2321</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $tissue_hires_scalef</span></span>
<span><span class="co">#&gt; [1] 0.1220703</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $tissue_lowres_scalef</span></span>
<span><span class="co">#&gt; [1] 0.03662109</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $spot_diameter_um</span></span>
<span><span class="co">#&gt; [1] 55</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $pixels_per_um</span></span>
<span><span class="co">#&gt; [1] 2.367856</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="create-spatial-scmetalink-object">Create Spatial scMetaLink Object<a class="anchor" aria-label="anchor" href="#create-spatial-scmetalink-object"></a>
</h2>
<p>Use <code><a href="../reference/createScMetaLinkFromSpatial.html">createScMetaLinkFromSpatial()</a></code> to create an object
with spatial coordinates.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create spatial scMetaLink object</span></span>
<span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createScMetaLinkFromSpatial.html">createScMetaLinkFromSpatial</a></span><span class="op">(</span></span>
<span>  expression_data <span class="op">=</span> <span class="va">st_expr</span>,</span>
<span>  spatial_coords <span class="op">=</span> <span class="va">st_meta</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"x"</span>, <span class="st">"y"</span><span class="op">)</span><span class="op">]</span>,</span>
<span>  cell_meta <span class="op">=</span> <span class="va">st_meta</span>,</span>
<span>  cell_type_column <span class="op">=</span> <span class="st">"cell_type"</span>,</span>
<span>  scale_factors <span class="op">=</span> <span class="va">st_scalefactors</span>,  <span class="co"># Important for correct distance calculation!</span></span>
<span>  min_cells <span class="op">=</span> <span class="fl">5</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check the object</span></span>
<span><span class="va">obj</span></span>
<span><span class="co">#&gt; scMetaLink Object</span></span>
<span><span class="co">#&gt; =================</span></span>
<span><span class="co">#&gt; Genes: 4284</span></span>
<span><span class="co">#&gt; Cells: 1000</span></span>
<span><span class="co">#&gt; Cell types: 6 (Endothelial, Stromal, Fibroblast, ...)</span></span></code></pre></div>
<div class="section level3">
<h3 id="important-coordinate-units">Important: Coordinate Units<a class="anchor" aria-label="anchor" href="#important-coordinate-units"></a>
</h3>
<p>For <strong>10x Visium</strong> data:</p>
<ul>
<li>Raw coordinates are in <strong>pixels</strong>, not micrometers</li>
<li>You <strong>must</strong> provide <code>scale_factors</code> with
<code>pixels_per_um</code> for correct distance calculations</li>
<li>Without proper scaling, distance parameters (sigma, threshold) will
be interpreted as pixels!</li>
</ul>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Check spatial distance statistics</span></span>
<span><span class="va">dist_stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getSpatialDistanceStats.html">getSpatialDistanceStats</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"=== Spatial Distance Statistics ===\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; === Spatial Distance Statistics ===</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Number of spots:"</span>, <span class="va">dist_stats</span><span class="op">$</span><span class="va">n_spots</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Number of spots: 1000</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Min distance:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">dist_stats</span><span class="op">$</span><span class="va">min_distance_um</span>, <span class="fl">1</span><span class="op">)</span>, <span class="st">"um\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Min distance: 84.5 um</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Median distance:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">dist_stats</span><span class="op">$</span><span class="va">median_distance_um</span>, <span class="fl">1</span><span class="op">)</span>, <span class="st">"um\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Median distance: 2558.5 um</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Max distance:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">dist_stats</span><span class="op">$</span><span class="va">max_distance_um</span>, <span class="fl">1</span><span class="op">)</span>, <span class="st">"um\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Max distance: 6864.2 um</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Coordinate unit:"</span>, <span class="va">dist_stats</span><span class="op">$</span><span class="va">coord_unit</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Coordinate unit: converted to micrometers</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="working-with-seurat-objects">Working with Seurat Objects<a class="anchor" aria-label="anchor" href="#working-with-seurat-objects"></a>
</h3>
<p>If you have a Seurat spatial object, use
<code><a href="../reference/createScMetaLinkFromSeuratSpatial.html">createScMetaLinkFromSeuratSpatial()</a></code>:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># From Seurat spatial object</span></span>
<span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createScMetaLinkFromSeuratSpatial.html">createScMetaLinkFromSeuratSpatial</a></span><span class="op">(</span></span>
<span>  seurat_obj <span class="op">=</span> <span class="va">seurat_spatial</span>,</span>
<span>  cell_type_column <span class="op">=</span> <span class="st">"cell_type"</span>,  <span class="co"># Column with deconvolution results</span></span>
<span>  assay <span class="op">=</span> <span class="st">"Spatial"</span>,</span>
<span>  image <span class="op">=</span> <span class="st">"slice1"</span>  <span class="co"># Name of the spatial image</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="visualize-spatial-cell-types">Visualize Spatial Cell Types<a class="anchor" aria-label="anchor" href="#visualize-spatial-cell-types"></a>
</h2>
<p>Before analysis, visualize the spatial distribution of cell
types.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotSpatialCellTypes.html">plotSpatialCellTypes</a></span><span class="op">(</span><span class="va">obj</span>, point_size <span class="op">=</span> <span class="fl">2</span>, alpha <span class="op">=</span> <span class="fl">0.8</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="spatial-analysis_files/figure-html/plot_celltypes-1.png" alt="**Figure 3: Spatial Cell Type Distribution.** Each spot is colored by its assigned cell type. For Visium data, cell types typically come from deconvolution methods (RCTD, cell2location, etc.)." width="100%"><p class="caption">
<strong>Figure 3: Spatial Cell Type Distribution.</strong> Each spot is
colored by its assigned cell type. For Visium data, cell types typically
come from deconvolution methods (RCTD, cell2location, etc.).
</p>
</div>
</div>
<div class="section level2">
<h2 id="infer-production-and-sensing">Infer Production and Sensing<a class="anchor" aria-label="anchor" href="#infer-production-and-sensing"></a>
</h2>
<p>Run the standard production and sensing inference.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Infer metabolite production</span></span>
<span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/inferProduction.html">inferProduction</a></span><span class="op">(</span></span>
<span>  <span class="va">obj</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"combined"</span>,</span>
<span>  consider_degradation <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  consider_secretion <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Infer metabolite sensing</span></span>
<span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/inferSensing.html">inferSensing</a></span><span class="op">(</span></span>
<span>  <span class="va">obj</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"combined"</span>,</span>
<span>  weight_by_affinity <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  include_transporters <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="visualize-spatial-metabolite-patterns">Visualize Spatial Metabolite Patterns<a class="anchor" aria-label="anchor" href="#visualize-spatial-metabolite-patterns"></a>
</h2>
<p>Visualize production and sensing scores on spatial coordinates.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotSpatialFeature.html">plotSpatialFeature</a></span><span class="op">(</span></span>
<span>  <span class="va">obj</span>,</span>
<span>  metabolite <span class="op">=</span> <span class="st">"L-Lactic acid"</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"production"</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">2</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="spatial-analysis_files/figure-html/plot_production-1.png" alt="**Figure 4: Lactate Production Potential.** Spots are colored by lactate production score based on their cell type. Warmer colors indicate higher production potential." width="100%"><p class="caption">
<strong>Figure 4: Lactate Production Potential.</strong> Spots are
colored by lactate production score based on their cell type. Warmer
colors indicate higher production potential.
</p>
</div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotSpatialFeature.html">plotSpatialFeature</a></span><span class="op">(</span></span>
<span>  <span class="va">obj</span>,</span>
<span>  metabolite <span class="op">=</span> <span class="st">"L-Lactic acid"</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"sensing"</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"#EBF5FB"</span>,</span>
<span>  high_color <span class="op">=</span> <span class="st">"#1A5276"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="spatial-analysis_files/figure-html/plot_sensing-1.png" alt="**Figure 5: Lactate Sensing Capability.** Spots are colored by lactate sensing score. This shows which regions can detect and respond to lactate signaling." width="100%"><p class="caption">
<strong>Figure 5: Lactate Sensing Capability.</strong> Spots are colored
by lactate sensing score. This shows which regions can detect and
respond to lactate signaling.
</p>
</div>
<div class="section level3">
<h3 id="compare-multiple-metabolites">Compare Multiple Metabolites<a class="anchor" aria-label="anchor" href="#compare-multiple-metabolites"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotSpatialComparison.html">plotSpatialComparison</a></span><span class="op">(</span></span>
<span>  <span class="va">obj</span>,</span>
<span>  metabolites <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"L-Lactic acid"</span>, <span class="st">"L-Glutamic acid"</span>, <span class="st">"Adenosine"</span>, <span class="st">"L-Alanine"</span><span class="op">)</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"production"</span>,</span>
<span>  ncol <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">1.5</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="spatial-analysis_files/figure-html/compare_mets-1.png" alt="**Figure 6: Spatial Comparison of Key Metabolites.** Production patterns for multiple metabolites shown side by side, revealing distinct spatial organization of metabolic activities." width="100%"><p class="caption">
<strong>Figure 6: Spatial Comparison of Key Metabolites.</strong>
Production patterns for multiple metabolites shown side by side,
revealing distinct spatial organization of metabolic activities.
</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="compute-spatial-communication">Compute Spatial Communication<a class="anchor" aria-label="anchor" href="#compute-spatial-communication"></a>
</h2>
<p>The key innovation is <strong>spatially-weighted
communication</strong>. Nearby cell type pairs receive higher
communication scores.</p>
<div class="section level3">
<h3 id="spatial-weighting-methods">Spatial Weighting Methods<a class="anchor" aria-label="anchor" href="#spatial-weighting-methods"></a>
</h3>
<p>scMetaLink offers several spatial weighting methods:</p>
<table class="table">
<thead><tr class="header">
<th>Method</th>
<th>Formula</th>
<th>Best for</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>knn</code></td>
<td>K-nearest neighbors only</td>
<td>Visium (recommended)</td>
</tr>
<tr class="even">
<td><code>gaussian</code></td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>e</mi><mrow><mi>−</mi><msup><mi>d</mi><mn>2</mn></msup><mi>/</mi><mn>2</mn><msup><mi>σ</mi><mn>2</mn></msup></mrow></msup><annotation encoding="application/x-tex">e^{-d^2/2\sigma^2}</annotation></semantics></math></td>
<td>Smooth decay</td>
</tr>
<tr class="odd">
<td><code>exponential</code></td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msup><mi>e</mi><mrow><mi>−</mi><mi>d</mi><mi>/</mi><mi>λ</mi></mrow></msup><annotation encoding="application/x-tex">e^{-d/\lambda}</annotation></semantics></math></td>
<td>Sharp decay</td>
</tr>
<tr class="even">
<td><code>linear</code></td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mrow><mi>max</mi><mo>⁡</mo></mrow><mo stretchy="false" form="prefix">(</mo><mn>0</mn><mo>,</mo><mn>1</mn><mo>−</mo><mi>d</mi><mi>/</mi><msub><mi>d</mi><mrow><mi>m</mi><mi>a</mi><mi>x</mi></mrow></msub><mo stretchy="false" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">\max(0, 1 - d/d_{max})</annotation></semantics></math></td>
<td>Simple model</td>
</tr>
<tr class="odd">
<td><code>threshold</code></td>
<td>Binary cutoff</td>
<td>Strict distance limit</td>
</tr>
</tbody>
</table>
<div class="figure" style="text-align: center">
<img src="spatial-analysis_files/figure-html/weighting-1.png" alt="**Figure 7: Spatial Weighting Functions.** Different methods model the decay of communication potential with distance. KNN (not shown) uses a binary approach based on nearest neighbors." width="100%"><p class="caption">
<strong>Figure 7: Spatial Weighting Functions.</strong> Different
methods model the decay of communication potential with distance. KNN
(not shown) uses a binary approach based on nearest neighbors.
</p>
</div>
</div>
<div class="section level3">
<h3 id="run-spatial-communication-analysis">Run Spatial Communication Analysis<a class="anchor" aria-label="anchor" href="#run-spatial-communication-analysis"></a>
</h3>
<p>For <strong>Visium data</strong>, we recommend the <code>knn</code>
method with k=6 (hexagonal grid neighbors):</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Compute spatially-weighted communication</span></span>
<span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/computeSpatialCommunication.html">computeSpatialCommunication</a></span><span class="op">(</span></span>
<span>  <span class="va">obj</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"knn"</span>,           <span class="co"># K-nearest neighbors (recommended for Visium)</span></span>
<span>  k_neighbors <span class="op">=</span> <span class="fl">6</span>,          <span class="co"># 6 neighbors for hexagonal grid</span></span>
<span>  symmetric <span class="op">=</span> <span class="cn">TRUE</span>,         <span class="co"># Bidirectional communication potential</span></span>
<span>  comm_method <span class="op">=</span> <span class="st">"geometric"</span>, <span class="co"># sqrt(production x sensing)</span></span>
<span>  min_production <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  min_sensing <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  n_permutations <span class="op">=</span> <span class="fl">100</span>,     <span class="co"># Use &gt;=1000 for publication</span></span>
<span>  n_cores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  seed <span class="op">=</span> <span class="fl">42</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt;   |                                                                              |                                                                      |   0%  |                                                                              |=                                                                     |   1%  |                                                                              |=                                                                     |   2%  |                                                                              |==                                                                    |   3%  |                                                                              |===                                                                   |   4%  |                                                                              |====                                                                  |   5%  |                                                                              |====                                                                  |   6%  |                                                                              |=====                                                                 |   7%  |                                                                              |======                                                                |   8%  |                                                                              |======                                                                |   9%  |                                                                              |=======                                                               |  10%  |                                                                              |========                                                              |  11%  |                                                                              |========                                                              |  12%  |                                                                              |=========                                                             |  13%  |                                                                              |==========                                                            |  14%  |                                                                              |==========                                                            |  15%  |                                                                              |===========                                                           |  16%  |                                                                              |============                                                          |  17%  |                                                                              |=============                                                         |  18%  |                                                                              |=============                                                         |  19%  |                                                                              |==============                                                        |  20%  |                                                                              |===============                                                       |  21%  |                                                                              |===============                                                       |  22%  |                                                                              |================                                                      |  23%  |                                                                              |=================                                                     |  24%  |                                                                              |==================                                                    |  25%  |                                                                              |==================                                                    |  26%  |                                                                              |===================                                                   |  27%  |                                                                              |====================                                                  |  28%  |                                                                              |====================                                                  |  29%  |                                                                              |=====================                                                 |  30%  |                                                                              |======================                                                |  31%  |                                                                              |======================                                                |  32%  |                                                                              |=======================                                               |  33%  |                                                                              |========================                                              |  34%  |                                                                              |========================                                              |  35%  |                                                                              |=========================                                             |  36%  |                                                                              |==========================                                            |  37%  |                                                                              |===========================                                           |  38%  |                                                                              |===========================                                           |  39%  |                                                                              |============================                                          |  40%  |                                                                              |=============================                                         |  41%  |                                                                              |=============================                                         |  42%  |                                                                              |==============================                                        |  43%  |                                                                              |===============================                                       |  44%  |                                                                              |================================                                      |  45%  |                                                                              |================================                                      |  46%  |                                                                              |=================================                                     |  47%  |                                                                              |==================================                                    |  48%  |                                                                              |==================================                                    |  49%  |                                                                              |===================================                                   |  50%  |                                                                              |====================================                                  |  51%  |                                                                              |====================================                                  |  52%  |                                                                              |=====================================                                 |  53%  |                                                                              |======================================                                |  54%  |                                                                              |======================================                                |  55%  |                                                                              |=======================================                               |  56%  |                                                                              |========================================                              |  57%  |                                                                              |=========================================                             |  58%  |                                                                              |=========================================                             |  59%  |                                                                              |==========================================                            |  60%  |                                                                              |===========================================                           |  61%  |                                                                              |===========================================                           |  62%  |                                                                              |============================================                          |  63%  |                                                                              |=============================================                         |  64%  |                                                                              |==============================================                        |  65%  |                                                                              |==============================================                        |  66%  |                                                                              |===============================================                       |  67%  |                                                                              |================================================                      |  68%  |                                                                              |================================================                      |  69%  |                                                                              |=================================================                     |  70%  |                                                                              |==================================================                    |  71%  |                                                                              |==================================================                    |  72%  |                                                                              |===================================================                   |  73%  |                                                                              |====================================================                  |  74%  |                                                                              |====================================================                  |  75%  |                                                                              |=====================================================                 |  76%  |                                                                              |======================================================                |  77%  |                                                                              |=======================================================               |  78%  |                                                                              |=======================================================               |  79%  |                                                                              |========================================================              |  80%  |                                                                              |=========================================================             |  81%  |                                                                              |=========================================================             |  82%  |                                                                              |==========================================================            |  83%  |                                                                              |===========================================================           |  84%  |                                                                              |============================================================          |  85%  |                                                                              |============================================================          |  86%  |                                                                              |=============================================================         |  87%  |                                                                              |==============================================================        |  88%  |                                                                              |==============================================================        |  89%  |                                                                              |===============================================================       |  90%  |                                                                              |================================================================      |  91%  |                                                                              |================================================================      |  92%  |                                                                              |=================================================================     |  93%  |                                                                              |==================================================================    |  94%  |                                                                              |==================================================================    |  95%  |                                                                              |===================================================================   |  96%  |                                                                              |====================================================================  |  97%  |                                                                              |===================================================================== |  98%  |                                                                              |===================================================================== |  99%  |                                                                              |======================================================================| 100%</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="alternative-gaussian-decay">Alternative: Gaussian Decay<a class="anchor" aria-label="anchor" href="#alternative-gaussian-decay"></a>
</h3>
<p>For more continuous distance weighting:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/computeSpatialCommunication.html">computeSpatialCommunication</a></span><span class="op">(</span></span>
<span>  <span class="va">obj</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"gaussian"</span>,</span>
<span>  sigma <span class="op">=</span> <span class="fl">50</span>,               <span class="co"># 50 um characteristic decay</span></span>
<span>  distance_threshold <span class="op">=</span> <span class="fl">150</span>, <span class="co"># Max communication distance</span></span>
<span>  n_permutations <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="filter-significant-interactions">Filter Significant Interactions<a class="anchor" aria-label="anchor" href="#filter-significant-interactions"></a>
</h2>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Filter significant spatial interactions</span></span>
<span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterSignificantInteractions.html">filterSignificantInteractions</a></span><span class="op">(</span></span>
<span>  <span class="va">obj</span>,</span>
<span>  pvalue_threshold <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  adjust_method <span class="op">=</span> <span class="st">"none"</span>  <span class="co"># Use "BH" for real analysis with more permutations</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Significant spatial interactions:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">obj</span><span class="op">@</span><span class="va">significant_interactions</span><span class="op">)</span>, <span class="st">"\n\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Significant spatial interactions: 442</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">obj</span><span class="op">@</span><span class="va">significant_interactions</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sender"</span>, <span class="st">"receiver"</span>, <span class="st">"metabolite_name"</span>,</span>
<span>                                       <span class="st">"communication_score"</span>, <span class="st">"pvalue_adjusted"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;       sender   receiver                 metabolite_name communication_score</span></span>
<span><span class="co">#&gt; 1    Stromal Fibroblast               Prostaglandin F1a           0.9781756</span></span>
<span><span class="co">#&gt; 2    Stromal    Stromal 20-Hydroxyeicosatetraenoic acid           0.9716861</span></span>
<span><span class="co">#&gt; 3 Fibroblast    Stromal       DG(18:0/18:2(9Z,12Z)/0:0)           0.9605915</span></span>
<span><span class="co">#&gt; 4    Stromal    Stromal                Prostaglandin I2           0.9602908</span></span>
<span><span class="co">#&gt; 5 Fibroblast    Stromal                         Retinal           0.9583222</span></span>
<span><span class="co">#&gt; 6 Fibroblast    Stromal               DG(18:0/18:0/0:0)           0.9579443</span></span>
<span><span class="co">#&gt;   pvalue_adjusted</span></span>
<span><span class="co">#&gt; 1      0.00990099</span></span>
<span><span class="co">#&gt; 2      0.00990099</span></span>
<span><span class="co">#&gt; 3      0.00990099</span></span>
<span><span class="co">#&gt; 4      0.01980198</span></span>
<span><span class="co">#&gt; 5      0.00990099</span></span>
<span><span class="co">#&gt; 6      0.00990099</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="visualize-spatial-communication-network">Visualize Spatial Communication Network<a class="anchor" aria-label="anchor" href="#visualize-spatial-communication-network"></a>
</h2>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotSpatialCommunicationNetwork.html">plotSpatialCommunicationNetwork</a></span><span class="op">(</span></span>
<span>  <span class="va">obj</span>,</span>
<span>  metabolite <span class="op">=</span> <span class="cn">NULL</span>,  <span class="co"># Aggregate all metabolites</span></span>
<span>  top_n <span class="op">=</span> <span class="fl">15</span>,</span>
<span>  arrow_scale <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">6</span>,</span>
<span>  show_labels <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="spatial-analysis_files/figure-html/network-1.png" alt="**Figure 8: Spatial Communication Network.** Arrows connect cell type centroids, with width proportional to communication strength. Background spots show the tissue context." width="100%"><p class="caption">
<strong>Figure 8: Spatial Communication Network.</strong> Arrows connect
cell type centroids, with width proportional to communication strength.
Background spots show the tissue context.
</p>
</div>
<div class="section level3">
<h3 id="metabolite-specific-network">Metabolite-Specific Network<a class="anchor" aria-label="anchor" href="#metabolite-specific-network"></a>
</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotSpatialCommunicationNetwork.html">plotSpatialCommunicationNetwork</a></span><span class="op">(</span></span>
<span>  <span class="va">obj</span>,</span>
<span>  metabolite <span class="op">=</span> <span class="st">"L-Lactic acid"</span>,</span>
<span>  top_n <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  arrow_scale <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">6</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="spatial-analysis_files/figure-html/network_met-1.png" alt="**Figure 9: Lactate Communication Network.** Spatial communication pattern for lactate specifically, showing which cell types communicate via this metabolite." width="100%"><p class="caption">
<strong>Figure 9: Lactate Communication Network.</strong> Spatial
communication pattern for lactate specifically, showing which cell types
communicate via this metabolite.
</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="analyze-communication-patterns">Analyze Communication Patterns<a class="anchor" aria-label="anchor" href="#analyze-communication-patterns"></a>
</h2>
<div class="section level3">
<h3 id="top-spatial-interactions">Top Spatial Interactions<a class="anchor" aria-label="anchor" href="#top-spatial-interactions"></a>
</h3>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sig</span> <span class="op">&lt;-</span> <span class="va">obj</span><span class="op">@</span><span class="va">significant_interactions</span></span>
<span></span>
<span><span class="co"># Top metabolites in spatial communication</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"=== Top Metabolite Mediators (Spatial) ===\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; === Top Metabolite Mediators (Spatial) ===</span></span>
<span><span class="va">met_counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">sig</span><span class="op">$</span><span class="va">metabolite_name</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="va">met_counts</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>, <span class="fl">15</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;                          24-Hydroxycholesterol </span></span>
<span><span class="co">#&gt;                                              5 </span></span>
<span><span class="co">#&gt;               DG(18:0/20:4(5Z,8Z,11Z,14Z)/0:0) </span></span>
<span><span class="co">#&gt;                                              5 </span></span>
<span><span class="co">#&gt;                                    Epinephrine </span></span>
<span><span class="co">#&gt;                                              5 </span></span>
<span><span class="co">#&gt;                                L-Aspartic acid </span></span>
<span><span class="co">#&gt;                                              5 </span></span>
<span><span class="co">#&gt;                               Prostaglandin D2 </span></span>
<span><span class="co">#&gt;                                              5 </span></span>
<span><span class="co">#&gt;                               Prostaglandin E2 </span></span>
<span><span class="co">#&gt;                                              5 </span></span>
<span><span class="co">#&gt;                                      Adenosine </span></span>
<span><span class="co">#&gt;                                              4 </span></span>
<span><span class="co">#&gt;                             Pentadecanoic acid </span></span>
<span><span class="co">#&gt;                                              4 </span></span>
<span><span class="co">#&gt;                                        13-HODE </span></span>
<span><span class="co">#&gt;                                              3 </span></span>
<span><span class="co">#&gt;                           13-cis-Retinoic acid </span></span>
<span><span class="co">#&gt;                                              3 </span></span>
<span><span class="co">#&gt;                 17-Hydroxypregnenolone sulfate </span></span>
<span><span class="co">#&gt;                                              3 </span></span>
<span><span class="co">#&gt;                        17a-Hydroxypregnenolone </span></span>
<span><span class="co">#&gt;                                              3 </span></span>
<span><span class="co">#&gt; 3 alpha,7 alpha,26-Trihydroxy-5beta-cholestane </span></span>
<span><span class="co">#&gt;                                              3 </span></span>
<span><span class="co">#&gt;                          3-Hydroxybutyric acid </span></span>
<span><span class="co">#&gt;                                              3 </span></span>
<span><span class="co">#&gt;                              5'-Deoxyadenosine </span></span>
<span><span class="co">#&gt;                                              3</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="cell-type-communication-summary">Cell Type Communication Summary<a class="anchor" aria-label="anchor" href="#cell-type-communication-summary"></a>
</h3>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Outgoing (sender) strength</span></span>
<span><span class="va">outgoing</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/aggregate.html" class="external-link">aggregate</a></span><span class="op">(</span><span class="va">communication_score</span> <span class="op">~</span> <span class="va">sender</span>, data <span class="op">=</span> <span class="va">sig</span>, FUN <span class="op">=</span> <span class="va">sum</span><span class="op">)</span></span>
<span><span class="va">outgoing</span> <span class="op">&lt;-</span> <span class="va">outgoing</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="op">-</span><span class="va">outgoing</span><span class="op">$</span><span class="va">communication_score</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html" class="external-link">barplot</a></span><span class="op">(</span><span class="va">outgoing</span><span class="op">$</span><span class="va">communication_score</span>, names.arg <span class="op">=</span> <span class="va">outgoing</span><span class="op">$</span><span class="va">sender</span>,</span>
<span>        las <span class="op">=</span> <span class="fl">2</span>, col <span class="op">=</span> <span class="st">"#E64B35"</span>, main <span class="op">=</span> <span class="st">"Outgoing Spatial Communication"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Incoming (receiver) strength</span></span>
<span><span class="va">incoming</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/aggregate.html" class="external-link">aggregate</a></span><span class="op">(</span><span class="va">communication_score</span> <span class="op">~</span> <span class="va">receiver</span>, data <span class="op">=</span> <span class="va">sig</span>, FUN <span class="op">=</span> <span class="va">sum</span><span class="op">)</span></span>
<span><span class="va">incoming</span> <span class="op">&lt;-</span> <span class="va">incoming</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="op">-</span><span class="va">incoming</span><span class="op">$</span><span class="va">communication_score</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html" class="external-link">barplot</a></span><span class="op">(</span><span class="va">incoming</span><span class="op">$</span><span class="va">communication_score</span>, names.arg <span class="op">=</span> <span class="va">incoming</span><span class="op">$</span><span class="va">receiver</span>,</span>
<span>        las <span class="op">=</span> <span class="fl">2</span>, col <span class="op">=</span> <span class="st">"#4DBBD5"</span>, main <span class="op">=</span> <span class="st">"Incoming Spatial Communication"</span><span class="op">)</span></span></code></pre></div>
<p><img src="spatial-analysis_files/figure-html/comm_summary-1.png" alt="" width="100%" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="communication-hotspot-identification">Communication Hotspot Identification<a class="anchor" aria-label="anchor" href="#communication-hotspot-identification"></a>
</h2>
<p>For spot-level analysis, scMetaLink can identify spatial hotspots of
communication activity.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Note: Requires running computeSpatialCommunication with analysis_level="spot"</span></span>
<span><span class="va">obj_spot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/computeSpatialCommunication.html">computeSpatialCommunication</a></span><span class="op">(</span></span>
<span>  <span class="va">obj</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"knn"</span>,</span>
<span>  k_neighbors <span class="op">=</span> <span class="fl">6</span>,</span>
<span>  analysis_level <span class="op">=</span> <span class="st">"spot"</span>,  <span class="co"># Spot-level analysis</span></span>
<span>  n_permutations <span class="op">=</span> <span class="fl">0</span>,       <span class="co"># Skip permutation for speed</span></span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Identify hotspots</span></span>
<span><span class="va">hotspots</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/identifyCommunicationHotspots.html">identifyCommunicationHotspots</a></span><span class="op">(</span></span>
<span>  <span class="va">obj_spot</span>,</span>
<span>  metabolite <span class="op">=</span> <span class="st">"L-Lactic acid"</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"sender"</span>,</span>
<span>  n_hotspots <span class="op">=</span> <span class="fl">5</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">hotspots</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Visualize hotspots</span></span>
<span><span class="fu"><a href="../reference/plotSpatialHotspots.html">plotSpatialHotspots</a></span><span class="op">(</span><span class="va">obj_spot</span>, metabolite <span class="op">=</span> <span class="st">"L-Lactic acid"</span>, type <span class="op">=</span> <span class="st">"sender"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="spatial-distance-distribution">Spatial Distance Distribution<a class="anchor" aria-label="anchor" href="#spatial-distance-distribution"></a>
</h2>
<p>Understand the spatial relationships in your data:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotSpatialDistanceDistribution.html">plotSpatialDistanceDistribution</a></span><span class="op">(</span><span class="va">obj</span>, max_distance <span class="op">=</span> <span class="fl">1000</span>, bins <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="spatial-analysis_files/figure-html/distance_dist-1.png" alt="**Figure 10: Spatial Distance Distribution.** Histogram showing the distribution of pairwise distances between spots. Red lines mark common distance thresholds (100, 200, 500 um)." width="100%"><p class="caption">
<strong>Figure 10: Spatial Distance Distribution.</strong> Histogram
showing the distribution of pairwise distances between spots. Red lines
mark common distance thresholds (100, 200, 500 um).
</p>
</div>
</div>
<div class="section level2">
<h2 id="parameter-guidelines">Parameter Guidelines<a class="anchor" aria-label="anchor" href="#parameter-guidelines"></a>
</h2>
<div class="section level3">
<h3 id="distance-thresholds-by-metabolite-type">Distance Thresholds by Metabolite Type<a class="anchor" aria-label="anchor" href="#distance-thresholds-by-metabolite-type"></a>
</h3>
<table class="table">
<thead><tr class="header">
<th>Metabolite Category</th>
<th>Typical Range</th>
<th>Recommended sigma</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>ATP, Adenosine</td>
<td>20-50 um</td>
<td>20-30 um</td>
</tr>
<tr class="even">
<td>Lactate</td>
<td>50-100 um</td>
<td>50-80 um</td>
</tr>
<tr class="odd">
<td>Amino acids</td>
<td>80-150 um</td>
<td>80-120 um</td>
</tr>
<tr class="even">
<td>Lipid mediators</td>
<td>100-200 um</td>
<td>100-150 um</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="platform-specific-recommendations">Platform-Specific Recommendations<a class="anchor" aria-label="anchor" href="#platform-specific-recommendations"></a>
</h3>
<table class="table">
<thead><tr class="header">
<th>Platform</th>
<th>Spot Size</th>
<th>Spacing</th>
<th>Recommended Method</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>10x Visium</td>
<td>55 um</td>
<td>~100 um</td>
<td>knn (k=6)</td>
</tr>
<tr class="even">
<td>Slide-seq</td>
<td>10 um</td>
<td>~10 um</td>
<td>gaussian (sigma=30)</td>
</tr>
<tr class="odd">
<td>MERFISH</td>
<td>subcellular</td>
<td>variable</td>
<td>gaussian (sigma=10-20)</td>
</tr>
<tr class="even">
<td>Stereo-seq</td>
<td>0.5 um</td>
<td>variable</td>
<td>knn or gaussian</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="complete-workflow-example">Complete Workflow Example<a class="anchor" aria-label="anchor" href="#complete-workflow-example"></a>
</h2>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Complete spatial analysis workflow</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Zaoqu-Liu/scMetaLink" class="external-link">scMetaLink</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 1. Load data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">st_expr</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">st_meta</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">st_scalefactors</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Create spatial object</span></span>
<span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createScMetaLinkFromSpatial.html">createScMetaLinkFromSpatial</a></span><span class="op">(</span></span>
<span>  expression_data <span class="op">=</span> <span class="va">st_expr</span>,</span>
<span>  spatial_coords <span class="op">=</span> <span class="va">st_meta</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"x"</span>, <span class="st">"y"</span><span class="op">)</span><span class="op">]</span>,</span>
<span>  cell_meta <span class="op">=</span> <span class="va">st_meta</span>,</span>
<span>  cell_type_column <span class="op">=</span> <span class="st">"cell_type"</span>,</span>
<span>  scale_factors <span class="op">=</span> <span class="va">st_scalefactors</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 3. Infer production and sensing</span></span>
<span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/inferProduction.html">inferProduction</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/inferSensing.html">inferSensing</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 4. Compute spatial communication</span></span>
<span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/computeSpatialCommunication.html">computeSpatialCommunication</a></span><span class="op">(</span></span>
<span>  <span class="va">obj</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"knn"</span>,</span>
<span>  k_neighbors <span class="op">=</span> <span class="fl">6</span>,</span>
<span>  n_permutations <span class="op">=</span> <span class="fl">1000</span>  <span class="co"># More permutations for publication</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 5. Filter significant interactions</span></span>
<span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterSignificantInteractions.html">filterSignificantInteractions</a></span><span class="op">(</span><span class="va">obj</span>, adjust_method <span class="op">=</span> <span class="st">"BH"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 6. Visualize results</span></span>
<span><span class="fu"><a href="../reference/plotSpatialCellTypes.html">plotSpatialCellTypes</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plotSpatialCommunicationNetwork.html">plotSpatialCommunicationNetwork</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 7. Export results</span></span>
<span><span class="fu"><a href="../reference/exportResults.html">exportResults</a></span><span class="op">(</span><span class="va">obj</span>, output_dir <span class="op">=</span> <span class="st">"spatial_results"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="key-differences-from-non-spatial-analysis">Key Differences from Non-Spatial Analysis<a class="anchor" aria-label="anchor" href="#key-differences-from-non-spatial-analysis"></a>
</h2>
<table class="table">
<colgroup>
<col width="26%">
<col width="43%">
<col width="30%">
</colgroup>
<thead><tr class="header">
<th>Aspect</th>
<th>Non-Spatial</th>
<th>Spatial</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><strong>Cell relationships</strong></td>
<td>All pairs equal</td>
<td>Distance-weighted</td>
</tr>
<tr class="even">
<td><strong>Communication score</strong></td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msqrt><mrow><mi>P</mi><mo>×</mo><mi>S</mi></mrow></msqrt><annotation encoding="application/x-tex">\sqrt{P \times S}</annotation></semantics></math></td>
<td><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msqrt><mrow><mi>P</mi><mo>×</mo><mi>S</mi></mrow></msqrt><mo>×</mo><mi>w</mi><mo stretchy="false" form="prefix">(</mo><mi>d</mi><mo stretchy="false" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">\sqrt{P \times S} \times w(d)</annotation></semantics></math></td>
</tr>
<tr class="odd">
<td><strong>Visualization</strong></td>
<td>Network/heatmap</td>
<td>Tissue coordinates</td>
</tr>
<tr class="even">
<td><strong>Biological meaning</strong></td>
<td>Potential communication</td>
<td>Likely communication</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="limitations-and-considerations">Limitations and Considerations<a class="anchor" aria-label="anchor" href="#limitations-and-considerations"></a>
</h2>
<ol style="list-style-type: decimal">
<li>
<strong>Resolution</strong>: Visium spots contain multiple cells;
cell type assignments are approximations</li>
<li>
<strong>Deconvolution</strong>: Use proper deconvolution methods
(RCTD, cell2location) for cell type assignment</li>
<li>
<strong>Diffusion assumptions</strong>: Metabolite diffusion is
simplified as distance-dependent decay</li>
<li>
<strong>3D tissue</strong>: Current analysis is 2D; vertical
diffusion is not modeled</li>
</ol>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.0 (2024-04-24)</span></span>
<span><span class="co">#&gt; Platform: aarch64-apple-darwin20</span></span>
<span><span class="co">#&gt; Running under: macOS 15.6.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] C</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Asia/Shanghai</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] Matrix_1.7-4     scMetaLink_1.0.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] gtable_0.3.6       jsonlite_2.0.0     dplyr_1.1.4        compiler_4.4.0    </span></span>
<span><span class="co">#&gt;  [5] tidyselect_1.2.1   dichromat_2.0-0.1  jquerylib_0.1.4    systemfonts_1.3.1 </span></span>
<span><span class="co">#&gt;  [9] scales_1.4.0       textshaping_1.0.4  yaml_2.3.12        fastmap_1.2.0     </span></span>
<span><span class="co">#&gt; [13] lattice_0.22-7     ggplot2_4.0.1      R6_2.6.1           labeling_0.4.3    </span></span>
<span><span class="co">#&gt; [17] generics_0.1.4     knitr_1.51         htmlwidgets_1.6.4  tibble_3.3.1      </span></span>
<span><span class="co">#&gt; [21] desc_1.4.3         bslib_0.9.0        pillar_1.11.1      RColorBrewer_1.1-3</span></span>
<span><span class="co">#&gt; [25] rlang_1.1.7        cachem_1.1.0       xfun_0.56          fs_1.6.6          </span></span>
<span><span class="co">#&gt; [29] sass_0.4.10        S7_0.2.1           otel_0.2.0         cli_3.6.5         </span></span>
<span><span class="co">#&gt; [33] withr_3.0.2        pkgdown_2.1.3      magrittr_2.0.4     digest_0.6.39     </span></span>
<span><span class="co">#&gt; [37] grid_4.4.0         lifecycle_1.0.5    vctrs_0.7.0        evaluate_1.0.5    </span></span>
<span><span class="co">#&gt; [41] glue_1.8.0         farver_2.1.2       ragg_1.5.0         rmarkdown_2.30    </span></span>
<span><span class="co">#&gt; [45] tools_4.4.0        pkgconfig_2.0.3    htmltools_0.5.9</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Zaoqu Liu.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
