<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Quick Start Guide • scMetaLink</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Quick Start Guide">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">scMetaLink</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/quick-start.html">Quick Start Guide</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/theory.html">Theory &amp; Methods</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/production-sensing.html">Production &amp; Sensing Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/communication.html">Communication Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/spatial-analysis.html">Spatial Transcriptomics</a></li>
    <li><a class="dropdown-item" href="../articles/lactate-signaling.html">Lactate Signaling</a></li>
    <li><a class="dropdown-item" href="../articles/visualization.html">Visualization Gallery</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../articles/applications.html">Application Examples</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/Zaoqu-Liu/scMetaLink" aria-label="GitHub"><span class="fa fab fa-github"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Quick Start Guide</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/Zaoqu-Liu/scMetaLink/blob/main/vignettes/quick-start.Rmd" class="external-link"><code>vignettes/quick-start.Rmd</code></a></small>
      <div class="d-none name"><code>quick-start.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><strong>scMetaLink</strong> infers metabolite-mediated cell-cell
communication from single-cell RNA-seq data. Unlike traditional
ligand-receptor analysis, scMetaLink focuses on metabolites as signaling
mediators, capturing a crucial but often overlooked layer of
intercellular communication.</p>
<div class="figure" style="text-align: center">
<img src="flowchart.png" class="r-plt" alt="**Figure 1: scMetaLink Workflow.** Overview of the scMetaLink analysis pipeline." width="100%"><p class="caption">
<strong>Figure 1: scMetaLink Workflow.</strong> Overview of the
scMetaLink analysis pipeline.
</p>
</div>
<div class="section level3">
<h3 id="what-youll-learn">What You’ll Learn<a class="anchor" aria-label="anchor" href="#what-youll-learn"></a>
</h3>
<ol style="list-style-type: decimal">
<li>How to set up a scMetaLink analysis</li>
<li>The basic workflow from data input to results</li>
<li>How to interpret the main outputs</li>
</ol>
</div>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install from GitHub</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"Zaoqu-Liu/scMetaLink"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-package-and-example-data">Load Package and Example Data<a class="anchor" aria-label="anchor" href="#load-package-and-example-data"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Zaoqu-Liu/scMetaLink" class="external-link">scMetaLink</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://Matrix.R-forge.R-project.org" class="external-link">Matrix</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load built-in colorectal cancer example data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">crc_example</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check the data structure</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Expression matrix dimensions:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">crc_expr</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Expression matrix dimensions: 4210 2850</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Genes:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">crc_expr</span><span class="op">)</span>, <span class="st">"| Cells:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">crc_expr</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Genes: 4210 | Cells: 2850</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"\nCell metadata:\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Cell metadata:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">crc_meta</span><span class="op">)</span></span>
<span><span class="co">#&gt;                   cell_type tumor_normal   tissue_region</span></span>
<span><span class="co">#&gt; Cell_0001 Normal Epithelial       Normal Normal Adjacent</span></span>
<span><span class="co">#&gt; Cell_0002 Normal Epithelial       Normal Normal Adjacent</span></span>
<span><span class="co">#&gt; Cell_0003 Normal Epithelial       Normal Normal Adjacent</span></span>
<span><span class="co">#&gt; Cell_0004 Normal Epithelial       Normal Normal Adjacent</span></span>
<span><span class="co">#&gt; Cell_0005 Normal Epithelial       Normal Normal Adjacent</span></span>
<span><span class="co">#&gt; Cell_0006 Normal Epithelial       Normal Normal Adjacent</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"\nCell type distribution:\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Cell type distribution:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">crc_meta</span><span class="op">$</span><span class="va">cell_type</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;                 B               CAF       Endothelial          Gliacyte </span></span>
<span><span class="co">#&gt;               150               200               100                20 </span></span>
<span><span class="co">#&gt;              Mast          Monocyte Normal Epithelial Normal Fibroblast </span></span>
<span><span class="co">#&gt;                30               120               300               200 </span></span>
<span><span class="co">#&gt; Normal Macrophage          Pericyte            Plasma               SMC </span></span>
<span><span class="co">#&gt;               150                50               250                30 </span></span>
<span><span class="co">#&gt;                 T               TAM  Tumor Epithelial </span></span>
<span><span class="co">#&gt;               500               150               600</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="the-5-minute-workflow">The 5-Minute Workflow<a class="anchor" aria-label="anchor" href="#the-5-minute-workflow"></a>
</h2>
<div class="section level3">
<h3 id="step-1-create-scmetalink-object">Step 1: Create scMetaLink Object<a class="anchor" aria-label="anchor" href="#step-1-create-scmetalink-object"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create the analysis object</span></span>
<span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createScMetaLink.html">createScMetaLink</a></span><span class="op">(</span></span>
<span>  expression_data <span class="op">=</span> <span class="va">crc_expr</span>,</span>
<span>  cell_meta <span class="op">=</span> <span class="va">crc_meta</span>,</span>
<span>  cell_type_column <span class="op">=</span> <span class="st">"cell_type"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check the object</span></span>
<span><span class="va">obj</span></span>
<span><span class="co">#&gt; scMetaLink Object</span></span>
<span><span class="co">#&gt; =================</span></span>
<span><span class="co">#&gt; Genes: 4210</span></span>
<span><span class="co">#&gt; Cells: 2850</span></span>
<span><span class="co">#&gt; Cell types: 15 (Normal Epithelial, T, Plasma, ...)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-2-infer-metabolite-production">Step 2: Infer Metabolite Production<a class="anchor" aria-label="anchor" href="#step-2-infer-metabolite-production"></a>
</h3>
<p>Production potential reflects how capable each cell type is at
<strong>synthesizing and secreting</strong> specific metabolites.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/inferProduction.html">inferProduction</a></span><span class="op">(</span></span>
<span>  <span class="va">obj</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"combined"</span>, <span class="co"># Use mean expression x proportion expressing</span></span>
<span>  consider_degradation <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># Subtract degradation enzyme expression</span></span>
<span>  consider_secretion <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># Weight by secretion potential</span></span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View top lactate producers</span></span>
<span><span class="fu"><a href="../reference/getTopProducers.html">getTopProducers</a></span><span class="op">(</span><span class="va">obj</span>, <span class="st">"L-Lactic acid"</span>, top_n <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="co">#&gt;           cell_type production_score rank</span></span>
<span><span class="co">#&gt; 1          Monocyte        0.9375074    1</span></span>
<span><span class="co">#&gt; 2          Gliacyte        0.5305421    2</span></span>
<span><span class="co">#&gt; 3               TAM        0.5021831    3</span></span>
<span><span class="co">#&gt; 4 Normal Macrophage        0.4832566    4</span></span>
<span><span class="co">#&gt; 5  Tumor Epithelial        0.3845369    5</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-3-infer-metabolite-sensing">Step 3: Infer Metabolite Sensing<a class="anchor" aria-label="anchor" href="#step-3-infer-metabolite-sensing"></a>
</h3>
<p>Sensing capability reflects how capable each cell type is at
<strong>detecting and responding to</strong> specific metabolites.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/inferSensing.html">inferSensing</a></span><span class="op">(</span></span>
<span>  <span class="va">obj</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"combined"</span>,</span>
<span>  weight_by_affinity <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># Weight by receptor-metabolite affinity</span></span>
<span>  include_transporters <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># Include uptake transporters</span></span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View top glutamate sensors</span></span>
<span><span class="fu"><a href="../reference/getTopSensors.html">getTopSensors</a></span><span class="op">(</span><span class="va">obj</span>, <span class="st">"L-Glutamic acid"</span>, top_n <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="co">#&gt;           cell_type sensing_score rank</span></span>
<span><span class="co">#&gt; 1 Normal Macrophage     0.7624215    1</span></span>
<span><span class="co">#&gt; 2               TAM     0.6192208    2</span></span>
<span><span class="co">#&gt; 3  Tumor Epithelial     0.4362282    3</span></span>
<span><span class="co">#&gt; 4          Gliacyte     0.4255517    4</span></span>
<span><span class="co">#&gt; 5 Normal Fibroblast     0.4229089    5</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-4-compute-communication">Step 4: Compute Communication<a class="anchor" aria-label="anchor" href="#step-4-compute-communication"></a>
</h3>
<p>Communication scores quantify the potential signal flow from sender
(producer) to receiver (sensor) cells.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/computeCommunication.html">computeCommunication</a></span><span class="op">(</span></span>
<span>  <span class="va">obj</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"geometric"</span>, <span class="co"># sqrt(production x sensing)</span></span>
<span>  min_production <span class="op">=</span> <span class="fl">0.1</span>, <span class="co"># Filter weak producers</span></span>
<span>  min_sensing <span class="op">=</span> <span class="fl">0.1</span>, <span class="co"># Filter weak sensors</span></span>
<span>  n_permutations <span class="op">=</span> <span class="fl">100</span>, <span class="co"># Permutation test for significance</span></span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt;   |                                                                              |                                                                      |   0%  |                                                                              |=                                                                     |   1%  |                                                                              |=                                                                     |   2%  |                                                                              |==                                                                    |   3%  |                                                                              |===                                                                   |   4%  |                                                                              |====                                                                  |   5%  |                                                                              |====                                                                  |   6%  |                                                                              |=====                                                                 |   7%  |                                                                              |======                                                                |   8%  |                                                                              |======                                                                |   9%  |                                                                              |=======                                                               |  10%  |                                                                              |========                                                              |  11%  |                                                                              |========                                                              |  12%  |                                                                              |=========                                                             |  13%  |                                                                              |==========                                                            |  14%  |                                                                              |==========                                                            |  15%  |                                                                              |===========                                                           |  16%  |                                                                              |============                                                          |  17%  |                                                                              |=============                                                         |  18%  |                                                                              |=============                                                         |  19%  |                                                                              |==============                                                        |  20%  |                                                                              |===============                                                       |  21%  |                                                                              |===============                                                       |  22%  |                                                                              |================                                                      |  23%  |                                                                              |=================                                                     |  24%  |                                                                              |==================                                                    |  25%  |                                                                              |==================                                                    |  26%  |                                                                              |===================                                                   |  27%  |                                                                              |====================                                                  |  28%  |                                                                              |====================                                                  |  29%  |                                                                              |=====================                                                 |  30%  |                                                                              |======================                                                |  31%  |                                                                              |======================                                                |  32%  |                                                                              |=======================                                               |  33%  |                                                                              |========================                                              |  34%  |                                                                              |========================                                              |  35%  |                                                                              |=========================                                             |  36%  |                                                                              |==========================                                            |  37%  |                                                                              |===========================                                           |  38%  |                                                                              |===========================                                           |  39%  |                                                                              |============================                                          |  40%  |                                                                              |=============================                                         |  41%  |                                                                              |=============================                                         |  42%  |                                                                              |==============================                                        |  43%  |                                                                              |===============================                                       |  44%  |                                                                              |================================                                      |  45%  |                                                                              |================================                                      |  46%  |                                                                              |=================================                                     |  47%  |                                                                              |==================================                                    |  48%  |                                                                              |==================================                                    |  49%  |                                                                              |===================================                                   |  50%  |                                                                              |====================================                                  |  51%  |                                                                              |====================================                                  |  52%  |                                                                              |=====================================                                 |  53%  |                                                                              |======================================                                |  54%  |                                                                              |======================================                                |  55%  |                                                                              |=======================================                               |  56%  |                                                                              |========================================                              |  57%  |                                                                              |=========================================                             |  58%  |                                                                              |=========================================                             |  59%  |                                                                              |==========================================                            |  60%  |                                                                              |===========================================                           |  61%  |                                                                              |===========================================                           |  62%  |                                                                              |============================================                          |  63%  |                                                                              |=============================================                         |  64%  |                                                                              |==============================================                        |  65%  |                                                                              |==============================================                        |  66%  |                                                                              |===============================================                       |  67%  |                                                                              |================================================                      |  68%  |                                                                              |================================================                      |  69%  |                                                                              |=================================================                     |  70%  |                                                                              |==================================================                    |  71%  |                                                                              |==================================================                    |  72%  |                                                                              |===================================================                   |  73%  |                                                                              |====================================================                  |  74%  |                                                                              |====================================================                  |  75%  |                                                                              |=====================================================                 |  76%  |                                                                              |======================================================                |  77%  |                                                                              |=======================================================               |  78%  |                                                                              |=======================================================               |  79%  |                                                                              |========================================================              |  80%  |                                                                              |=========================================================             |  81%  |                                                                              |=========================================================             |  82%  |                                                                              |==========================================================            |  83%  |                                                                              |===========================================================           |  84%  |                                                                              |============================================================          |  85%  |                                                                              |============================================================          |  86%  |                                                                              |=============================================================         |  87%  |                                                                              |==============================================================        |  88%  |                                                                              |==============================================================        |  89%  |                                                                              |===============================================================       |  90%  |                                                                              |================================================================      |  91%  |                                                                              |================================================================      |  92%  |                                                                              |=================================================================     |  93%  |                                                                              |==================================================================    |  94%  |                                                                              |==================================================================    |  95%  |                                                                              |===================================================================   |  96%  |                                                                              |====================================================================  |  97%  |                                                                              |===================================================================== |  98%  |                                                                              |===================================================================== |  99%  |                                                                              |======================================================================| 100%</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-5-filter-significant-interactions">Step 5: Filter Significant Interactions<a class="anchor" aria-label="anchor" href="#step-5-filter-significant-interactions"></a>
</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># For this quick tutorial with limited permutations, we use no adjustment</span></span>
<span><span class="co"># In real analysis with more permutations, use adjust_method = "BH"</span></span>
<span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterSignificantInteractions.html">filterSignificantInteractions</a></span><span class="op">(</span></span>
<span>  <span class="va">obj</span>,</span>
<span>  pvalue_threshold <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  adjust_method <span class="op">=</span> <span class="st">"none"</span> <span class="co"># Use "BH" for real analysis</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View top significant interactions</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">obj</span><span class="op">@</span><span class="va">significant_interactions</span>, <span class="fl">10</span><span class="op">)</span></span>
<span><span class="co">#&gt;    metabolite_id            sender receiver communication_score     pvalue</span></span>
<span><span class="co">#&gt; 1    HMDB0000068          Monocyte     Mast           0.9999641 0.00990099</span></span>
<span><span class="co">#&gt; 2    HMDB0003818 Normal Epithelial   Plasma           0.9992733 0.00990099</span></span>
<span><span class="co">#&gt; 3    HMDB0002685       Endothelial     Mast           0.9903007 0.00990099</span></span>
<span><span class="co">#&gt; 4    HMDB0004234          Gliacyte        B           0.9862472 0.00990099</span></span>
<span><span class="co">#&gt; 5    HMDB0000153            Plasma   Plasma           0.9822572 0.00990099</span></span>
<span><span class="co">#&gt; 6    HMDB0000653 Normal Macrophage        T           0.9768051 0.00990099</span></span>
<span><span class="co">#&gt; 7    HMDB0013078  Tumor Epithelial        B           0.9747729 0.00990099</span></span>
<span><span class="co">#&gt; 8    HMDB0000190          Monocyte        B           0.9679184 0.00990099</span></span>
<span><span class="co">#&gt; 9    HMDB0001139       Endothelial      CAF           0.9659777 0.00990099</span></span>
<span><span class="co">#&gt; 10   HMDB0000335  Tumor Epithelial   Plasma           0.9555982 0.00990099</span></span>
<span><span class="co">#&gt;    pvalue_adjusted        metabolite_name</span></span>
<span><span class="co">#&gt; 1       0.00990099            Epinephrine</span></span>
<span><span class="co">#&gt; 2       0.00990099       5-Androstenediol</span></span>
<span><span class="co">#&gt; 3       0.00990099      Prostaglandin F1a</span></span>
<span><span class="co">#&gt; 4       0.00990099 12-Keto-leukotriene B4</span></span>
<span><span class="co">#&gt; 5       0.00990099                Estriol</span></span>
<span><span class="co">#&gt; 6       0.00990099    Cholesterol sulfate</span></span>
<span><span class="co">#&gt; 7       0.00990099   Stearoylethanolamide</span></span>
<span><span class="co">#&gt; 8       0.00990099          L-Lactic acid</span></span>
<span><span class="co">#&gt; 9       0.00990099      Prostaglandin F2a</span></span>
<span><span class="co">#&gt; 10      0.00990099     16a-Hydroxyestrone</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="quick-visualization">Quick Visualization<a class="anchor" aria-label="anchor" href="#quick-visualization"></a>
</h2>
<div class="section level3">
<h3 id="communication-heatmap">Communication Heatmap<a class="anchor" aria-label="anchor" href="#communication-heatmap"></a>
</h3>
<p>The heatmap shows pairwise communication strength between cell types,
aggregated across all significant metabolites.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotCommunicationHeatmap.html">plotCommunicationHeatmap</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="quick-start_files/figure-html/heatmap-1.png" class="r-plt" alt="**Figure 1: Communication Heatmap.** Each cell shows the total communication score from sender (rows) to receiver (columns) cell types. Darker colors indicate stronger communication." width="100%"><p class="caption">
<strong>Figure 1: Communication Heatmap.</strong> Each cell shows the
total communication score from sender (rows) to receiver (columns) cell
types. Darker colors indicate stronger communication.
</p>
</div>
</div>
<div class="section level3">
<h3 id="chord-diagram">Chord Diagram<a class="anchor" aria-label="anchor" href="#chord-diagram"></a>
</h3>
<p>The chord diagram provides an intuitive view of communication flow
between cell types.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plotCommunicationCircle.html">plotCommunicationCircle</a></span><span class="op">(</span><span class="va">obj</span>, top_n <span class="op">=</span> <span class="fl">50</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="quick-start_files/figure-html/chord-1.png" class="r-plt" alt="**Figure 2: Communication Chord Diagram.** Ribbons connect sender to receiver cell types, with width proportional to communication strength. Colors represent sender cell types." width="100%"><p class="caption">
<strong>Figure 2: Communication Chord Diagram.</strong> Ribbons connect
sender to receiver cell types, with width proportional to communication
strength. Colors represent sender cell types.
</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="one-line-workflow">One-Line Workflow<a class="anchor" aria-label="anchor" href="#one-line-workflow"></a>
</h2>
<p>For convenience, you can run the entire analysis with a single
function:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Complete analysis in one line</span></span>
<span><span class="va">obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/runScMetaLink.html">runScMetaLink</a></span><span class="op">(</span></span>
<span>  expression_data <span class="op">=</span> <span class="va">crc_expr</span>,</span>
<span>  cell_meta <span class="op">=</span> <span class="va">crc_meta</span>,</span>
<span>  cell_type_column <span class="op">=</span> <span class="st">"cell_type"</span>,</span>
<span>  n_permutations <span class="op">=</span> <span class="fl">100</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="understanding-the-output">Understanding the Output<a class="anchor" aria-label="anchor" href="#understanding-the-output"></a>
</h2>
<div class="section level3">
<h3 id="key-objects-in-the-result">Key Objects in the Result<a class="anchor" aria-label="anchor" href="#key-objects-in-the-result"></a>
</h3>
<table class="table">
<colgroup>
<col width="31%">
<col width="68%">
</colgroup>
<thead><tr class="header">
<th>Slot</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>@production_scores</code></td>
<td>Matrix: metabolites x cell types, production potential</td>
</tr>
<tr class="even">
<td><code>@sensing_scores</code></td>
<td>Matrix: metabolites x cell types, sensing capability</td>
</tr>
<tr class="odd">
<td><code>@communication_scores</code></td>
<td>3D array: sender x receiver x metabolite</td>
</tr>
<tr class="even">
<td><code>@communication_pvalues</code></td>
<td>3D array: permutation p-values</td>
</tr>
<tr class="odd">
<td><code>@significant_interactions</code></td>
<td>data.frame: filtered significant interactions</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="accessing-results">Accessing Results<a class="anchor" aria-label="anchor" href="#accessing-results"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Production scores (first 5 metabolites, first 5 cell types)</span></span>
<span><span class="va">obj</span><span class="op">@</span><span class="va">production_scores</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="co">#&gt;             Normal Epithelial         T    Plasma Normal Fibroblast       CAF</span></span>
<span><span class="co">#&gt; HMDB0000008         0.3844017 0.3124838 0.4670759         0.3124838 0.3124838</span></span>
<span><span class="co">#&gt; HMDB0000010         0.2708579 0.2465175 0.2391890         0.5269140 0.5625056</span></span>
<span><span class="co">#&gt; HMDB0000011         0.2804398 0.3580385 0.3628140         0.5658166 0.4273138</span></span>
<span><span class="co">#&gt; HMDB0000012         0.4768788 0.4473201 0.4021738         0.3852586 0.4177708</span></span>
<span><span class="co">#&gt; HMDB0000014         0.3271932 0.4098426 0.4522973         0.3559737 0.3869064</span></span>
<span></span>
<span><span class="co"># Significant interactions summary</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Total significant interactions:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">obj</span><span class="op">@</span><span class="va">significant_interactions</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Total significant interactions: 2754</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"\nTop metabolites involved:\n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Top metabolites involved:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">obj</span><span class="op">@</span><span class="va">significant_interactions</span><span class="op">$</span><span class="va">metabolite_name</span><span class="op">)</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 11beta-Hydroxyprogesterone           5-Androstenediol </span></span>
<span><span class="co">#&gt;                         30                         23 </span></span>
<span><span class="co">#&gt;                  Ornithine   7beta-Hydroxycholesterol </span></span>
<span><span class="co">#&gt;                         21                         20 </span></span>
<span><span class="co">#&gt;    Adenosine monophosphate                  Cortisone </span></span>
<span><span class="co">#&gt;                         19                         19 </span></span>
<span><span class="co">#&gt;                  Hydroxide                    Retinal </span></span>
<span><span class="co">#&gt;                         19                         18 </span></span>
<span><span class="co">#&gt;                   L-Fucose     Uridine 5'-diphosphate </span></span>
<span><span class="co">#&gt;                         17                         17</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="export-results">Export Results<a class="anchor" aria-label="anchor" href="#export-results"></a>
</h2>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Export all results to files</span></span>
<span><span class="fu"><a href="../reference/exportResults.html">exportResults</a></span><span class="op">(</span><span class="va">obj</span>, output_dir <span class="op">=</span> <span class="st">"scMetaLink_results"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="next-steps">Next Steps<a class="anchor" aria-label="anchor" href="#next-steps"></a>
</h2>
<ul>
<li>
<strong><a href="theory.html">Theory &amp; Methods</a></strong>:
Understand the mathematical framework</li>
<li>
<strong><a href="production-sensing.html">Production &amp;
Sensing</a></strong>: Deep dive into inference</li>
<li>
<strong><a href="communication.html">Communication
Analysis</a></strong>: Advanced communication analysis</li>
<li>
<strong><a href="spatial-analysis.html">Spatial
Analysis</a></strong>: Spatial transcriptomics support</li>
<li>
<strong><a href="visualization.html">Visualization</a></strong>:
Complete visualization guide</li>
<li>
<strong><a href="applications.html">Applications</a></strong>:
Real-world analysis examples</li>
</ul>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.5.2 (2025-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.3 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] Matrix_1.7-4     scMetaLink_1.0.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] viridis_0.6.5         sass_0.4.10           generics_0.1.4       </span></span>
<span><span class="co">#&gt;  [4] shape_1.4.6.1         lattice_0.22-7        digest_0.6.39        </span></span>
<span><span class="co">#&gt;  [7] magrittr_2.0.4        evaluate_1.0.5        grid_4.5.2           </span></span>
<span><span class="co">#&gt; [10] RColorBrewer_1.1-3    iterators_1.0.14      circlize_0.4.17      </span></span>
<span><span class="co">#&gt; [13] fastmap_1.2.0         foreach_1.5.2         doParallel_1.0.17    </span></span>
<span><span class="co">#&gt; [16] jsonlite_2.0.0        GlobalOptions_0.1.3   gridExtra_2.3        </span></span>
<span><span class="co">#&gt; [19] ComplexHeatmap_2.26.0 viridisLite_0.4.2     scales_1.4.0         </span></span>
<span><span class="co">#&gt; [22] codetools_0.2-20      textshaping_1.0.4     jquerylib_0.1.4      </span></span>
<span><span class="co">#&gt; [25] cli_3.6.5             rlang_1.1.7           crayon_1.5.3         </span></span>
<span><span class="co">#&gt; [28] cachem_1.1.0          yaml_2.3.12           otel_0.2.0           </span></span>
<span><span class="co">#&gt; [31] tools_4.5.2           parallel_4.5.2        dplyr_1.1.4          </span></span>
<span><span class="co">#&gt; [34] colorspace_2.1-2      ggplot2_4.0.1         BiocGenerics_0.56.0  </span></span>
<span><span class="co">#&gt; [37] GetoptLong_1.1.0      vctrs_0.7.0           R6_2.6.1             </span></span>
<span><span class="co">#&gt; [40] png_0.1-8             stats4_4.5.2          matrixStats_1.5.0    </span></span>
<span><span class="co">#&gt; [43] lifecycle_1.0.5       S4Vectors_0.48.0      IRanges_2.44.0       </span></span>
<span><span class="co">#&gt; [46] fs_1.6.6              htmlwidgets_1.6.4     clue_0.3-66          </span></span>
<span><span class="co">#&gt; [49] cluster_2.1.8.1       ragg_1.5.0            pkgconfig_2.0.3      </span></span>
<span><span class="co">#&gt; [52] desc_1.4.3            pkgdown_2.2.0         pillar_1.11.1        </span></span>
<span><span class="co">#&gt; [55] bslib_0.9.0           gtable_0.3.6          glue_1.8.0           </span></span>
<span><span class="co">#&gt; [58] systemfonts_1.3.1     xfun_0.56             tibble_3.3.1         </span></span>
<span><span class="co">#&gt; [61] tidyselect_1.2.1      knitr_1.51            farver_2.1.2         </span></span>
<span><span class="co">#&gt; [64] rjson_0.2.23          htmltools_0.5.9       rmarkdown_2.30       </span></span>
<span><span class="co">#&gt; [67] compiler_4.5.2        S7_0.2.1</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Zaoqu Liu.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
